roles/apache/tasks/main.yml

```diff
@@ -0,0 +1,59 @@
+---
+- name: install apache and necessary mods
+  apt: pkg={{ item }}
+  with_items:
+    - apache2
+    - libapache2-mod-wsgi
+    - libapache2-mod-proxy-html
+
+- name: install ssl cert+key
+  template: src=etc/apache2/{{ item }} dest=/etc/apache2/{{ item }}
+            owner=haproxy group=haproxy mode=0600
+  notify: reload apache
+  with_items:
+    - ssl.key
+    - ssl.crt
+    - ssl.conf
+
+- name: enable apache modules
+  apache2_module: state=present name={{item}}
+  notify:
+    - reload apache
+  with_items:
+    - wsgi
+    - ssl
+    - proxy
+    - proxy_ajp
+    - proxy_http
+    - rewrite
+    - deflate
+    - headers
+    - proxy_balancer
+    - proxy_connect
+    - proxy_html
+
+- name: disable apache status
+  apache2_module: state=absent name=status
+  notify:
+    - reload apache
+
+- name: disable default apache site
+  apache2_site: state=disabled name=000-default
+  notify:
+    - reload apache
+
+- name: check if custom ports.conf
+  exit_status: grep '# This file is managed by Ansible' /etc/apache2/ports.conf
```

Comment:
can we not use exit_status? It's a dep of antsy, and ruby, and it's custom, and we have to carry it forward, and we can achieve the same thing with ansible built ins such as failed_when